### DISCUSSION TITLE
Isn't nested mapping supported?

### DISCUSSION THREAD

**Discussion by tim-pankoke** [nartc/mapper#404] [Q&A] [answered]
Hi,

first of all: Thanks for this amazing library! Having a .net background it's a pleasure 😅

I'm wondering if nested mapping isn't supported or I'm doing something wrong 🤔

Let's say I have the following classes:

```typescript
class Tenant {
  @AutoMap()
  id: string;

  @AutoMap()
  name: string;
}

class User {
  @AutoMap()
  id: string;

  @AutoMap({ typeFn: () => Tenant })
  tenant: Tenant;
}

class UserModel {
  @AutoMap()
  id: string;

  @AutoMap()
  tenantId: string;

  @AutoMap()
  tenantName: string;
}
```

I want to map `User` into `UserModel` as well as `UserModel` into `User`.
Mapping `User` into `UserModel` works as expected and the flattening is done by automapper. (e.g. `tenant.id` -> `tenantId`)
But mapping `UserModel` into `User` results in an empty `tenant` property. (e.g. `tenantId` -> `tenant.id`)

I was able to fix this by specifying the mapping manually but would have expected that this works out-of-the-box.

Is that the expected behaviour?

Here is a CodeSandbox that demonstrates the behaviour: https://codesandbox.io/s/automapper-y6xoj?file=/src/index.ts

**Comment by nartc** [comment, marked as answer]
@tim-hoffmann Thank you for the kind words. Yes, unfortunately, this is the expected behavior. Reverse Mapping flattened fields back into objects has been proving very difficult to do :(. Manual mapping is the way to go here sadly.

**Reply by nartc** [reply]
If you have any suggestions of a solution, I'm all ears

**Reply by tim-pankoke** [reply]
Thank you for clarifying that!
